packages/@okta/vuepress-site/docs/concepts/interaction-code/index.md

```diff
@@ -0,0 +1,96 @@
+
+The Interaction Code grant is intended for developers who want a step-by-step remediation user experience without redirecting to an Authorization Server. This grant type enables developers to include [Identity Engine features](https://help.okta.com/en/oie/okta_help_CSH.htm#csh-features), such as passwordless authentication and progressive profiling. See [Redirect authentication vs. embedded authentication](/docs/concepts/redirect-vs-embedded/) for Identity Engine authentication deployment models and [Identity Engine deployment guides](/docs/guides/oie-intro/) for detailed deployment use cases.
+
+## The Interaction Code flow
+
+The Interaction Code flow is similar to the [OAuth 2.0 Authorization Code flow with PKCE](/docs/concepts/oauth-openid/#authorization-code-flow-with-pkce). All clients are required to pass along a client ID, as well as a Proof Key for Code Exchange (PKCE), to keep the flow secure. Confidential clients such as web apps must also pass a client secret in their authorization request. The user can start the authorization request with minimal information, relying on the client to facilitate the interactions with the Okta Authorization Server to progressively authenticate the user. The series of interactions, which could include multifactor authentication steps, is secured using the `interaction_handle`. After successfully completing the remedial interactions, the client receives an `interaction_code` that they can then redeem for tokens with the [`/token`](/docs/reference/api/oidc/#token) endpoint.
+
+The following table describes the parameters introduced for the Interaction Code grant type flow:
+
+| Interaction Code grant parameter           | Description   |
+| --------------------------------           | -----------   |
+| `interaction_handle` | The `interaction_handle` is an opaque, immutable value that is provided by the Okta Authorization Server. The client can use the `interaction_handle` to interact with the Okta Authorization Server directly. The client is responsible for saving the `interaction_handle` and using it for the duration of the transaction. Any public/confidential client that is configured to use the Interaction Code grant type can obtain an `interaction_handle`. If all the remediation steps are successfully performed, an `interaction_code` is returned as part of the success response.            |
+| `interaction_code` |  The `interaction_code` is a one-time use, opaque code that the client can exchange for tokens using the Interaction Code grant type. This code enables a client to redeem a completed Identity Engine interaction for tokens without needing access to an Authorization Server’s session. |
```

Comment:
is there a requirement when we need to use `Okta Authorization Server` or `Authorization Server`? Curious if having the same everywhere will be better for enduser.

Reply:
I will rework this part: will refer to generic AS as "authorization server", and refer to Okta's AS as "Org Authorization Server" or "Custom Authorization Server".

On that note, should we distinguish the Interaction Code interaction between "Authorization Server (Okta)" and "Identity Engine (Okta)" - from the sequence diagram? Since we are not discussing the transition from OAuth API to IDX API, should we just clump all the Interaction Code interaction to "Authorization Server (Okta)" and drop the "Identity Engine (Okta)" entity? Would that over simplify the interaction?

Reply:
👍 for the first paragraph.

And for the second part - we can drop the `Identity Engine (Okta)` as a separate component, but may be we can still include it within the `Authorization Server (Okta)` component in the way that is specifies that AS includes `Identity Engine` component.